src/network/transaction_request/mod.rs

```diff
@@ -18,75 +19,182 @@ pub struct TransactionRequest {
         self
     }
 
+    /// Get the paymaster params for the transaction.
+    pub fn paymaster_params(&self) -> Option<&PaymasterParams> {
+        self.eip_712_meta
+            .as_ref()
+            .and_then(|meta| meta.paymaster_params.as_ref())
+    }
+
+    /// Set the paymaster params  for the transaction.
+    pub fn set_paymaster_params(&mut self, paymaster_params: PaymasterParams) {
+        self.eip_712_meta
+            .get_or_insert_with(Eip712Meta::default)
+            .paymaster_params = Some(paymaster_params);
+    }
+
+    /// Builder-pattern method for setting paymaster params.
+    pub fn with_paymaster_params(mut self, paymaster_params: PaymasterParams) -> Self {
+        self.set_paymaster_params(paymaster_params);
+        self
+    }
+
+    /// Builder-pattern method for building a Zksync EIP-712 create2 tranasaction.
+    pub fn with_create2_params(
+        self,
+        salt: B256,
+        code: Vec<u8>,
+        constructor_data: Vec<u8>,
+        factory_deps: Vec<Vec<u8>>,
+    ) -> Result<Self, BytecodeHashError> {
+        self.with_create_params_inner(Some(salt), code, constructor_data, factory_deps)
+    }
+
+    /// Builder-pattern method for building a ZKsync EIP-712 create tranasaction.
+    pub fn with_create_params(
+        self,
+        code: Vec<u8>,
+        constructor_data: Vec<u8>,
+        factory_deps: Vec<Vec<u8>>,
+    ) -> Result<Self, BytecodeHashError> {
+        self.with_create_params_inner(None, code, constructor_data, factory_deps)
+    }
+
+    pub fn with_create_params_inner(
+        self,
+        salt: Option<B256>,
+        code: Vec<u8>,
+        constructor_data: Vec<u8>,
+        factory_deps: Vec<Vec<u8>>,
+    ) -> Result<Self, BytecodeHashError> {
+        let bytecode_hash = hash_bytecode(&code)?;
+        let factory_deps = factory_deps
+            .into_iter()
+            .chain(vec![code])
+            .map(Into::into)
+            .collect();
+        let input = crate::contracts::l2::contract_deployer::encode_create_calldata(
+            salt,
+            bytecode_hash.into(),
+            constructor_data.into(),
+        );
+        Ok(self
+            .with_to(CONTRACT_DEPLOYER_ADDRESS)
+            .with_input(input)
+            .with_factory_deps(factory_deps))
+    }
+}
+
+impl TransactionRequest {
+    #[deprecated(note = "use `set_paymaster_params` instead")]
+    pub fn set_paymaster(&mut self, paymaster_params: PaymasterParams) {
+        self.eip_712_meta
+            .get_or_insert_with(Eip712Meta::default)
+            .paymaster_params = Some(paymaster_params);
+    }
+
+    #[deprecated(note = "use `with_paymaster_params` instead")]
+    pub fn with_paymaster(mut self, paymaster_params: PaymasterParams) -> Self {
+        #[allow(deprecated)]
+        self.set_paymaster(paymaster_params);
+        self
+    }
+
+    #[deprecated(note = "use `with_create_params` instead")]
     pub fn zksync_deploy(
```

Comment:
The reason why I named the function this way is because it doesn't _only_ set params, it also changes the `to` field, which IMHO is not really represented here. Do you think the new name is more declarative?

Reply:
Under a `ZksyncTransactionBuilder` is was implicit as we can only deploy to calls via `CONTRACT_DEPLOYER_ADDRESS`, in fact most of the foundry-zksync code has been adapted in paces of `create`  to setup the `to` address. If we go with the axiom, `alloy_zksync::TransactionRequest` is Zksync EIP-712 by default, this makes sense, as the user would otherwise always need to set the `to` manually.

Reply:
alt: `build_create_transaction`,  `build_create2_transaction` ?

Reply:
I don't think `build` is applicable here, because `build` is typically the last method in the builder pattern that _finishes_ the build process, which is not the case here.